### PR TITLE
implement SingleAttestation

### PR DESCRIPTION
**Pull request by tersec**
Required for https://notes.ethereum.org/@ethpandaops/pectra-devnet-5

### CODE REVIEW

File: beacon_chain/gossip_processing/gossip_validation.nim [hunk 1]
```diff
@@ -1023,31 +1031,57 @@ proc validateAttestation*(
...
   # In the spec, is_valid_indexed_attestation is used to verify the signature -
   # here, we do a batch verification instead
-  let sig =
-    attestation.signature.load().valueOr:
-      return pool.checkedReject("Attestation: unable to load signature")
+  let sig = attestation.signature.load().valueOr:
+    return pool.checkedReject("SingleAttestation: unable to load signature")
 
-  return ok((validator_index, sig))
+  echo "FOOBAR4: ", beacon_committee.len, "; ", index_in_committee
```
Comment (arnetheduck) [inline beacon_chain/gossip_processing/gossip_validation.nim]: ..

File: beacon_chain/gossip_processing/gossip_validation.nim [hunk 2]
```diff
@@ -1023,31 +1031,57 @@ proc validateAttestation*(
...
+  # attestation.attester_index in
+  # get_beacon_committee(state, attestation.data.slot, index).
   let
-    shufflingRef =
-      pool.dag.getShufflingRef(target.blck, target.slot.epoch, false).valueOr:
-        # Target is verified - shouldn't happen
-        warn "No shuffling for attestation - report bug",
-          attestation = shortLog(attestation), target = shortLog(target)
-        return errIgnore("Attestation: no shuffling")
+    beacon_committee = get_beacon_committee(
```
Comment (arnetheduck) [inline beacon_chain/gossip_processing/gossip_validation.nim:1106]: here, ideally we'd have a fast and a slow path - the fast path is for when the shuffling is cached - then we check beacon committee and other "fast" checks and finally sig.. 

if instead the committee is unknown, we first check the signature and only then do we compute the shuffling to check the other fields

Comment (tersec) [reply beacon_chain/gossip_processing/gossip_validation.nim:1106]: https://github.com/status-im/nimbus-eth2/pull/6783/commits/659d03d7f291513e6049ca7839dd9fb420d116ab

File: beacon_chain/gossip_processing/gossip_validation.nim (hunk 1 above)
Comment (tersec) [reply beacon_chain/gossip_processing/gossip_validation.nim]: https://github.com/status-im/nimbus-eth2/pull/6783/commits/659d03d7f291513e6049ca7839dd9fb420d116ab

File: beacon_chain/gossip_processing/gossip_validation.nim [hunk 3]
```diff
@@ -1023,31 +1031,68 @@ proc validateAttestation*(
   # defined by attestation.data.beacon_block_root -- i.e.
   # get_checkpoint_block(store, attestation.data.beacon_block_root,
   # store.finalized_checkpoint.epoch) == store.finalized_checkpoint.root
+
+  var sig: CookedSig
+  let shufflingRefMissing =
```
Comment (arnetheduck) [inline beacon_chain/gossip_processing/gossip_validation.nim]: why not:

```nim
var sigchecked = false
let shufflingRef = findShufflingRef(...).valueOr:
  # check sig
  sigchecked = true
  getShufflingRefRef(...)
```

avoids the double lookup in the majority (happy) case

**Review by arnetheduck — APPROVED** [review]

File: beacon_chain/gossip_processing/gossip_validation.nim (hunk 3 above)
Comment (tersec) [reply beacon_chain/gossip_processing/gossip_validation.nim]: https://github.com/status-im/nimbus-eth2/pull/6783/commits/836b28f6429226019723b5b4e7775124c1924cca

**Pull request merged**